### PR TITLE
Link directly to document-download-api

### PR DESCRIPTION
**Pull request by sastels**
# Summary | Résumé

Currently if a file link is sent, it sends a link to document-download-frontend. There there is a redirect to a link to document-download-api. This PR skips the redirect and sends the document-download-api link directly. This means we no longer are using document-download-frontend.

Instead of an end user receiving an email of the form

>Message with file
https://document.notification.canada.ca/d/8xaqazjdTZ6D-LbuPh1Cjw/WcZuoQ97S8-kO--c1OwgUg?key=tjpAB4tLb-AzNLF8eaA6ukrFjdJepjzSjdmNkuIjLsQ&filename=test_file.txt
They will get one similar to

>Message with file
https://api.document.notification.canada.ca/services/a4d0638a-d06b-4971-9346-2f584f5b4ad5/documents/d1485b3d-591b-4048-bcdc-30e400a52ac0?key=dtoMOqOvH14b-uXtMGxlXErP8Fu_523U3dXErzAwoPY&sending_method=link&filename=test_file.txt

Difference is the host (`document.notification.canada.ca` vs `api.document.notification.canada.ca`) and the first link is of the format
```
host/base64(service_id)/base64(document_id)?key=...&filename=...
```
and the second is
```
host/services/service_id/documents/document_id?key=...&filename=...
```
If desired we could get rid of "services" and "documents" in the new link
```
host/service_id/document_id?key=...&filename=...
```
or do this as well as base64 encode the id's as we do in the old link.

Thoughts? I lean towards removing "services" and "documents" and leaving the ids there.

# Test instructions | Instructions pour tester la modification

See [How to test sending files](https://docs.google.com/document/d/1HsSVpctY8A1Py7On1og16chO25QrHYLq_IzxOARikhg), except don't run document-download-frontend.

# Release Instructions | Instructions pour le déploiement

# Reviewer checklist | Liste de vérification du réviseur

This is a suggested checklist of questions reviewers might ask during their
review | Voici une suggestion de liste de vérification comprenant des questions
que les réviseurs pourraient poser pendant leur examen :

- [ ] Is the code maintainable? | Est-ce que le code peut être maintenu?
- [ ] Have you tested it? | L’avez-vous testé?
- [ ] Are there automated tests? | Y a-t-il des tests automatisés?
- [ ] Does this cause automated test coverage to drop? | Est-ce que ça entraîne
      une baisse de la quantité de code couvert par les tests automatisés?
- [ ] Does this break existing functionality? | Est-ce que ça brise une
      fonctionnalité existante?
- [ ] Does this change the privacy policy? | Est-ce que ça entraîne une
      modification de la politique de confidentialité?
- [ ] Does this introduce any security concerns? | Est-ce que ça introduit des
      préoccupations liées à la sécurité?
- [ ] Does this significantly alter performance? | Est-ce que ça modifie de
      façon importante la performance?
- [ ] What is the risk level of using added dependencies? | Quel est le degré de
      risque d’utiliser des dépendances ajoutées?
- [ ] Should any documentation be updated as a result of this? (i.e. README
      setup, etc.) | Faudra-t-il mettre à jour la documentation à la suite de ce
      changement (fichier README, etc.)?

### CODE REVIEW

File: tests/app/delivery/test_send_to_providers.py [hunk 1]
```diff
@@ -978,8 +978,8 @@ def test_notification_with_bad_file_attachment_url(mocker, notify_db, notify_db_
     personalisation = {
         "file": document_download_response(
             {
-                "direct_file_url": "file://foo.bar/file.txt" if sending_method == "attach" else "http://foo.bar/file.txt",
-                "url": "file://foo.bar/file.txt" if sending_method == "link" else "http://foo.bar/file.txt",
+                "direct_file_url": "file://foo.bar/file.txt",
```
Comment (HEYGUL) [inline tests/app/delivery/test_send_to_providers.py:981]: `check_file_url` rejects any URL which does not start by `http`, thus for better clarity, I suggest following change:
```suggestion
                "direct_file_url": "http://foo.bar/file.txt",
```

**Comment by jzbahrai** [comment]
is this still relevant? either i can review or we can shut?

**Comment by jimleroyer** [comment]
@jzbahrai Still relevant. We reviewed PRs last week on Friday in this repo. We prefixed with prototype in the PR names for future work we want to test out.

File: tests/app/delivery/test_send_to_providers.py (hunk 1 above)
Comment (sastels) [reply tests/app/delivery/test_send_to_providers.py]: good point, thanks for the suggestion!

**Comment by sastels** [comment]
Closing and using a different approach: have `document-download-api` send the correct link rather than hacking it in here. See https://github.com/cds-snc/notification-document-download-api/pull/67

**Pull request closed without merging**